contracts/HundredVesting.sol

```diff
@@ -23,55 +23,50 @@ contract HundredVesting is Ownable {
         totalPeriod = totalPeriod / period;
     }
 
-    function getRemainedAmount (address vester) internal view returns (uint256) {
-        uint256 timestamp = addresses[vester].timestamp;
-        uint256 amount = (block.timestamp - timestamp) / timePerPeriod * addresses[vester].amount / totalPeriod;
-        return amount - addresses[vester].claimedAmount;
+    function getRemainedAmount (address vestedAddress) internal view returns (uint256) {
+        uint256 timestamp = addresses[vestedAddress].timestamp;
+        uint256 amount = (block.timestamp - timestamp) / timePerPeriod * addresses[vestedAddress].amount / totalPeriod - addresses[vestedAddress].claimedAmount;
+        return amount < 0 ? 0 : amount;
     }
     
-    function vesting (address vester, uint256 amount) external {
-        require(vester != address(0), "Invalid address");
+    function vesting (address vestedAddress, uint256 amount) external {
         require(amount != 0, "Amount should bigger than 0");
 
-        Hundred.transferFrom(msg.sender, address(this), amount);
-        UserInfo memory user = addresses[vester];
+        Hundred.safeTransferFrom(msg.sender, address(this), amount);
+        UserInfo memory user = addresses[vestedAddress];
         if (user.amount != 0) {
             user.amount = user.amount - user.claimedAmount + amount;
             user.timestamp = block.timestamp;
             user.claimedAmount = 0;
         } else {
             user = UserInfo({ amount: amount, timestamp: block.timestamp, claimedAmount: 0 });
         }
-        user.timestamp = block.timestamp;
-        addresses[vester] = user;
+        addresses[vestedAddress] = user;
     }
 
-    function claim (address vester) public {
-        require(vester != address(0), "Invalid address");
-        uint256 amount = getRemainedAmount(vester);
+    function claim () public {
+        require(msg.sender != address(0), "Invalid address");
+        uint256 amount = getRemainedAmount(msg.sender);
         require(amount != 0, "No claimable hundred token");
 
-        addresses[vester].claimedAmount = addresses[vester].claimedAmount + amount;
-        Hundred.transfer(msg.sender, amount);
+        Hundred.safeTransfer(msg.sender, amount);
+        addresses[msg.sender].claimedAmount = addresses[msg.sender].claimedAmount + amount;
     }
 
-    function getClaimableAmount (address vester) public view returns(uint) {
-        require(vester != address(0), "Invalid address");
-        UserInfo memory user = addresses[vester];
+    function getClaimableAmount () public view returns(uint) {
```

Comment:
this method does not do much in addition to calling `getRemainedAmount` so maybe there is no need to keep it